src/main.rs

```diff
@@ -108,7 +108,36 @@ fn main() {
     let _ = ansi_term::enable_ansi_support();
     logger::init();
 
-    let args = Cli::parse();
+    let args = match Cli::try_parse() {
+        Ok(args) => args,
+        Err(e) => {
+            let is_info_only = matches!(
+                e.kind,
+                clap::ErrorKind::DisplayHelpOnMissingArgumentOrSubcommand
+                    | clap::ErrorKind::DisplayHelp
+                    | clap::ErrorKind::DisplayVersion
+            );
+            // print the error
+            e.print().expect("error while writing error");
```

Comment:
It's better to avoid panicking here (#3280):
Upstream `parse` uses this:
```suggestion
            // avoid panic if stdout/stderr close unexpectedly
            let _ = e.print();
```

Reply:
thank you, will push asap